native_field/src/lib.rs

```diff
@@ -0,0 +1,266 @@
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec;
+use alloc::vec::Vec;
+use columns::{NativeFieldCols, COL_MAP, NUM_COLS};
+use core::mem::transmute;
+use valida_bus::{MachineWithGeneralBus, MachineWithRangeBus8};
+use valida_cpu::MachineWithCpuChip;
+use valida_machine::{instructions, Chip, Instruction, Interaction, Machine, Operands, Word};
+use valida_range::MachineWithRangeChip;
+
+use p3_air::VirtualPairCol;
+use p3_field::{Field, PrimeField32};
+use p3_matrix::dense::RowMajorMatrix;
+use p3_maybe_rayon::*;
+
+const ADD_OPCODE: u32 = 13;
+const SUB_OPCODE: u32 = 14;
+const MUL_OPCODE: u32 = 15;
+
+pub mod columns;
+pub mod stark;
+
+#[derive(Clone)]
+pub enum Operation {
+    Add(Word<u8>, Word<u8>, Word<u8>),
+    Sub(Word<u8>, Word<u8>, Word<u8>),
+    Mul(Word<u8>, Word<u8>, Word<u8>),
+}
+
+pub struct NativeFieldChip {
+    operations: Vec<Operation>,
+}
+
+impl<F, M> Chip<M> for NativeFieldChip
+where
+    F: Field,
+    M: MachineWithGeneralBus<F = F> + MachineWithRangeBus8,
+{
+    fn generate_trace(&self, _machine: &M) -> RowMajorMatrix<M::F> {
+        let rows = self
+            .operations
+            .par_iter()
+            .map(|op| self.op_to_row(op))
+            .collect::<Vec<_>>();
+
+        let mut trace =
+            RowMajorMatrix::new(rows.into_iter().flatten().collect::<Vec<_>>(), NUM_COLS);
+
+        Self::pad_to_power_of_two::<NUM_COLS, M::F>(&mut trace.values);
+
+        trace
+    }
+
+    fn global_sends(&self, machine: &M) -> Vec<Interaction<M::F>> {
+        let output = COL_MAP
+            .output
+            .0
+            .map(VirtualPairCol::single_main)
+            .into_iter()
+            .collect::<Vec<_>>();
+
+        let send = Interaction {
+            fields: output,
+            count: VirtualPairCol::one(),
```

Comment:
Should this be `is_add + is_sub + is_mul`?

I guess either way works, padding rows could do range checks too, it shouldn't really increase any costs. Just looks like we're not recording range checks for padding rows currently.